packages/@aws-cdk/aws-batch/lib/compute-environment.ts

```diff
@@ -0,0 +1,484 @@
+
+    // Only allow compute resources to be set when using UNMANAGED type
+    if (props.computeResources && !this.isManagedByAWS(props)) {
+      computeResources = {
+        allocationStrategy: props.allocationStrategy || AllocationStrategy.BEST_FIT,
+        bidPercentage: props.computeResources.bidPercentage,
+        desiredvCpus: props.computeResources.desiredvCpus,
+        ec2KeyPair: props.computeResources.ec2KeyPair,
+        imageId: props.computeResources.image && props.computeResources.image.getImage(this).imageId,
+        instanceRole: props.computeResources.instanceRole
+        ? props.computeResources.instanceRole.roleArn
+        : new iam.LazyRole(this, 'Resource-Instance-Role', {
+          assumedBy: new iam.ServicePrincipal('batch.amazonaws.com'),
+        }).roleArn,
+        instanceTypes: this.buildInstanceTypes(props.computeResources.instanceTypes),
+        launchTemplate: props.computeResources.launchTemplate,
+        maxvCpus: props.computeResources.maxvCpus || 256,
+        minvCpus: props.computeResources.minvCpus || 0,
+        placementGroup: props.computeResources.placementGroup && props.computeResources.placementGroup.ref,
+        securityGroupIds: this.buildSecurityGroupIds(props.computeResources.vpc, props.computeResources.securityGroups),
+        spotIamFleetRole: spotFleetRole ? spotFleetRole.roleArn : undefined,
+        subnets: props.computeResources.vpc.selectSubnets(props.computeResources.vpcSubnets).subnetIds,
+        tags: props.computeResources.computeResourcesTags,
+        type: props.computeResources.type || ComputeResourceType.ON_DEMAND,
+      };
+    }
+
+    const computeEnvironment = new CfnComputeEnvironment(this, 'Resource', {
+      computeEnvironmentName: this.physicalName,
+      computeResources,
+      serviceRole: props.serviceRole
+      ? props.serviceRole.roleArn
+      : new iam.LazyRole(this, 'Resource-Service-Instance-Role', {
+        managedPolicies: [
+          iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSBatchServiceRole'),
+        ],
+        assumedBy: new iam.ServicePrincipal('batch.amazonaws.com'),
+      }).roleArn,
+      state: this.isEnabled(props) ? 'ENABLED' : 'DISABLED',
+      type: this.isManagedByAWS(props) ? 'UNMANAGED' : 'MANAGED',
+    });
+
+    if (props.computeResources && props.computeResources.vpc) {
+      this.node.addDependency(props.computeResources.vpc);
+    }
+
+    this.computeEnvironmentArn = this.getResourceArnAttribute(computeEnvironment.ref, {
+      service: 'batch',
+      resource: 'compute-environment',
+      resourceName: this.physicalName,
+    });
+    this.computeEnvironmentName = this.getResourceNameAttribute(props.computeEnvironmentName || this.physicalName);
+  }
+
+  private isEnabled(props: ComputeEnvironmentProps): boolean {
+    return props.enabled === undefined ? true : props.enabled;
+  }
+
+  private isManagedByAWS(props: ComputeEnvironmentProps): boolean {
```

Comment:
The 'AWS' part of this name throws me off... can you change it to just `isManaged`?

Reply:
Got it 👍